php/controllers/Servlet.php

```diff
@@ -153,6 +169,67 @@ private function profile(){
     /**
      * Methods that actually do things
      */
+
+    private function changePassword(){
```

Comment:
returning "true" when the password and repeated password do not match?
If I run "changePassword()" from a different class, and see he result is "true", I will assume success.
Also, it is possible that your method simply does not return anything. An optional return is bad practice. 

Now this would be equal to writing a method saying
`changePassword() : boolean?` 
(? is the optional parameter), is best avoided. (I know PHP allows it, it's an issue with PHP, but bad practice ;-))

Either both "no $_POST" and ($newPassword != repeat) should return false, and success true, or return an error string and return the result, but not just nothing in one case, and a boolean in the other cases ;-)